runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/config/KafkaTopicTransformsConfigBuilder.java

```diff
@@ -89,11 +97,11 @@ public KafkaTopicTransformsConfigBuilder<T> extractHeader(
         if (matcher.reset(path).matches())
         {
             this.extractHeaders.add(new KafkaTopicHeaderType(name,
-                String.format(INTERNAL_VALUE, matcher.group(2))));
+                String.format(INTERNAL_VALUE, matcher.group(2)), externalPath));
         }
         else if (internalMatcher.reset(path).matches())
         {
-            this.extractHeaders.add(new KafkaTopicHeaderType(name, path));
+            this.extractHeaders.add(new KafkaTopicHeaderType(name, path, externalPath));
```

Comment:
When we build the `KafkaTopicTransformsConfig` there should be no matcher used, just capture the parsed values directly and nothing else.

In constructor of `KafkaTopicType`, where we prepare the `KafkaTopicConfig` for use in the binding for each core, that's where the parsed config transforms should be converted as needed to something like `KafkaTopicTransformType` using matcher etc, instead of using parsed transforms config directly.